lib/rules/no-irregular-whitespace.js

```diff
@@ -225,7 +225,7 @@ module.exports = {
 
                 // If we have any errors remaining report on them
                 errors.forEach(error => {
-                    context.report.apply(context, error);
+                    context.report(...error);
```

Comment:
I'm a bit confused here. Is `errors` an array of arrays? Maybe we're using old-style reporting here? Not sure how this should look.

Reply:
This rule seems to be doing a weird thing where it is storing problems as arrays of arguments, and doing old-style `context.report` calls on those arguments later. It would probably be a good idea to update this to use objects, but I didn't want to do it as part of this PR.